README.md

```diff
@@ -1 +1,197 @@
+[![Dependency Status](https://david-dm.org/libp2p/js-libp2p-circuit.svg?style=flat-square)](https://david-dm.org/libp2p/js-libp2p-circuit)
+[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/feross/standard)
+![](https://img.shields.io/badge/npm-%3E%3D3.0.0-orange.svg?style=flat-square)
+![](https://img.shields.io/badge/Node.js-%3E%3D4.0.0-orange.svg?style=flat-square)
+
+![](https://raw.githubusercontent.com/libp2p/interface-connection/master/img/badge.png)
+![](https://raw.githubusercontent.com/libp2p/interface-transport/master/img/badge.png)
+
+> Node.js implementation of the Circuit module that libp2p uses, which implements the [interface-connection](https://github.com/libp2p/interface-connection) interface for dial/listen.
+
+`libp2p-circuit` implements the circuit-relay mechanism that allows nodes that don't speak the same protocol to communicate using a third _relay_ node.
+
+**Note:** This module uses [pull-streams](https://pull-stream.github.io) for all stream based interfaces.
+
+### Why?
+
+`circuit-relaying` uses additional nodes in order to transfer traffic between two otherwise unreachable nodes. This allows nodes that don't speak the same protocols or are running in limited environments, e.g. browsers and IoT devices, to communicate, which would otherwise be impossible given the fact that for example browsers don't have any socket support and as such cannot be directly dialed.
+
+The applicability of circuit-relaying is not limited to routing traffic between browser nodes, other uses include:
+ - routing traffic between private nets
+ - circumventing NAT layers
+ - route mangling for better privacy (matreshka/shallot dialing).
+
+ It's also possible to use it for clients that implement exotic transports such as  devices that only have bluetooth radios to be reachable over bluethoot enabled relays and become full p2p nodes.
+
+### libp2p-circuit and IPFS
+
+Prior to `libp2p-circuit` there was a rift in the IPFS network, were IPFS nodes could only access content from nodes that speak the same protocol, for example TCP only nodes could only dial to other TCP only nodes, same for any other protocol combination. In practice, this limitation was most visible in JS-IPFS browser nodes, since they can only dial out but not be dialed in over WebRTC or WebSockets, hence any content that the browser node held was not reachable by the rest of the network even through it was announced on the DHT. Non browser IPFS nodes would usually speak more than one protocol such as TCP, WebSockets and/or WebRTC, this made the problem less severe outside of the browser. `libp2p-circuit` solves this problem completely, as long as there are `relay nodes` capable of routing traffic between those nodes their content should be available to the rest of the IPFS network.
```

Comment:
`TCP only nodes could only dial to other TCP only nodes` I don't think this is true (or badly worded at least). A node with a websocket and TCP transport could be dialed to from a TCP only node, or a websocket only node.

Reply:
Yeah, I was trying to convey the fact that _if only speaking one protocol_ nodes could be only dialed over that protocol. Awesome stuff tho... I need to really rework the readme.